### PR TITLE
Initial

### PR DESCRIPTION
**Pull request by thesoftwarejedi**
## Notes
Initial release

### CODE REVIEW

File: programs/step-staking/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,325 @@
...
+
+#[cfg(feature = "production")]
+declare_id!("UNKNOWN" fail build );
+#[cfg(not(feature = "production"))]
+declare_id!("STak7gf65TjoPaJvttZvinbgLy3vMMsB1ikDx1bK2mH");
+
+#[cfg(feature = "production")]
+const STEP_TOKEN_MINT_PUBKEY: &str = "StepAscQoEioFxxWGnh2sLBDFp9d8rvKz2Yp39iDpyT";
+#[cfg(not(feature = "production"))]
+const STEP_TOKEN_MINT_PUBKEY: &str = "sTEPVXgcctP7rJvoNk8p2Xmo1YrMbcMfu4tgHnowtFm";
```
Comment (aaronovz1) [inline programs/step-staking/src/lib.rs:16]: Shouldn't this be the same `StepAsc..` mint for mainnet and devnet?

File: programs/step-staking/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,325 @@
...
+
+#[cfg(feature = "production")]
+const X_STEP_TOKEN_MINT_PUBKEY: &str = "xStpgUCss9piqeFUk2iLVcvJEGhAdJxJQuwLkXP555G";
+#[cfg(not(feature = "production"))]
+const X_STEP_TOKEN_MINT_PUBKEY: &str = "xsTPvEj7rELYcqe2D1k3M5zRe85xWWFK3x1SWDN5qPY";
```
Comment (aaronovz1) [inline programs/step-staking/src/lib.rs:21]: This should be the same for mainnet and devnet too?

File: programs/step-staking/src/lib.rs (hunk 1 above)
Comment (thesoftwarejedi) [reply programs/step-staking/src/lib.rs:16]: yes.  the `not(feature = "production"))` is for unit testing

File: programs/step-staking/src/lib.rs (hunk 2 above)
Comment (thesoftwarejedi) [reply programs/step-staking/src/lib.rs:21]: yes.  the `not(feature = "production"))` is for unit testing

File: programs/step-staking/src/lib.rs (hunk 1 above)
Comment (thesoftwarejedi) [reply programs/step-staking/src/lib.rs:16]: changed to use feature named `local-testing`

**Pull request merged**